### PR TITLE
Add binding on the delete key to delete selected boxes

### PR DESCRIPTION
**Pull request by mattjgalloway**
As title. You can now press the backspace or delete key to delete items.

In this PR I also pull out the arrow key logic into a separate function. Gotta keep things sane in this file!

### CODE REVIEW

File: Views/js/designer.js [hunk 1]
```diff
@@ -494,62 +570,27 @@ var designer = {
...
+                        e.preventDefault();
+                    }
+                    break;
+                case 16: // Shift
+                    designer.shiftdown = true;
+                    break;
+                case 8:
+                case 46: // Backspace & delete
+                    if (designer.handle_delete_key_event(e)) {
+                        e.preventDefault();
```
Comment (Paul-Reed) [inline Views/js/designer.js]: Matt, the e.preventDefault(); doesn't appear to function for case 8 (backspace).
In Windows, if the backspace key is hit, the default action of restoring the previous page occurs and the item is not deleted.
This isn't a big issue, as most Windows users would I assume use the delete key anyway.

Paul

**Comment by Paul-Reed** [comment]
Matt, the e.preventDefault(); doesn't appear to function for case 8 (backspace).
In Windows, if the backspace key is hit, the default action of restoring the previous page occurs and the item is not deleted.
This isn't a big issue, as most Windows users would I assume use the delete key anyway.

Paul

**Comment by mattjgalloway** [comment]
Interesting. I guess that's OK though, if delete works for Windows.

**Comment by Paul-Reed** [comment]
Thanks Matt, looks a lot cleaner now.

**Pull request merged**